src/main/scala/gitbucket/core/service/RepositoryService.scala

```diff
@@ -301,6 +305,39 @@ trait RepositoryService {
       }
   }
 
+  def archiveRepository(owner: String, repository: String)(
+    implicit s: Session
+  ): Unit = {
+    Repositories
+      .filter(_.byRepository(owner, repository))
+      .map { t =>
+        t.isArchived
+      }
+      .update(true)
+  }
+
+  def unarchiveRepository(owner: String, repository: String)(
+    implicit s: Session
+  ): Unit = {
+    Repositories
+      .filter(_.byRepository(owner, repository))
+      .map { t =>
+        t.isArchived
+      }
+      .update(false)
+  }
+
+  def isArchivedRepository(owner: String, repository: String)(implicit s: Session): Option[Boolean] =
+    getRepository(owner, repository) match {
```

Comment:
This can be as follows:
```scala
getRepository(owner, repository).map { repository =>
  repository.repository.isArchived
}
```

Reply:
@takezoe fixed in https://github.com/gitbucket/gitbucket/pull/2595/commits/afe6e74b4cb34af6f56907c84e27883886a506bb